beep-beep/jobs.md

```diff
@@ -0,0 +1,1136 @@
+  PQ size is set to 100 and prefetch to 100000, you'll be able to push up to
+  prefetch number of jobs even if PQ is full.
+
+> Please note that this driver cannot have more than 1000 tasks with delay at 
+> the same time.
+
+### AMQP Driver
+
+Strictly speaking, AMQP is a protocol, not a full-fledged driver, so you can use
+any servers that support this protocol, such as: 
+[RabbitMQ](https://www.rabbitmq.com/), [Apache Qpid](http://qpid.apache.org/) or
+[Apache ActiveMQ](http://activemq.apache.org/). However, it is recommended to
+use RabbitMQ as the main implementation, and reliable performance with other 
+implementations is not guaranteed.
+
+To install and configure the RabbitMQ, use the corresponding
+[documentation page](https://www.rabbitmq.com/download.html). After that, you
+should configure the connection to the server in the "`amqp`" section. This
+configuration section contains exactly one `addr` key with a connection DSN
+in the format: `amqp://[USER]:[PASSWORD]@[HOST]:[PORT]`.
+
+```yaml
+amqp:
+  addr: amqp://guest:guest@localhost:5672
+```
+
+After creating a connection to the server, you can create a new queue that will
+use this connection and which will contain the queue settings (including 
+amqp-specific):
+
+```yaml
+amqp:
+  addr: amqp://guest:guest@localhost:5672
+
+
+jobs:
+  pipelines:
+    # User defined name of the queue.
+    example:
+      # Required section.
+      # Should be "amqp" for the AMQP driver.
+      driver: amqp
+      
+      # Optional section.
+      # Default: 10
+      priority: 10
+      
+      # Optional section.
+      # Default: 100
+      prefetch: 100
+
+      # Optional section.
+      # Default: "default"
+      queue: "default"
+
+      # Optional section.
+      # Default: "amqp.default"
+      exchange: "amqp.default"
+
+      # Optional section.
+      # Default: "direct"
+      exchange_type: "direct"
+
+      # Optional section.
+      # Default: "" (empty)
+      routing_key: ""
+
+      # Optional section.
+      # Default: false
+      exclusive: false
+
+      # Optional section.
+      # Default: false
+      multiple_ack: false
+
+      # Optional section.
+      # Default: false
+      requeue_on_fail: false
+```
+
+Below is a more detailed description of each of the amqp-specific options:
+- `priority` - Queue default priority for for each task pushed into this queue
+  if the priority value for these tasks was not explicitly set.
+
+- `prefetch` - The client can request that messages be sent in advance so that
+  when the client finishes processing a message, the following message is
+  already held locally, rather than needing to be sent down the channel.
+  Prefetching gives a performance improvement. This field specifies the prefetch
+  window size in octets. See also ["prefetch-size"](https://www.rabbitmq.com/amqp-0-9-1-reference.html) 
+  in AMQP QoS documentation reference.
+
+- `queue` - AMQP internal (inside the driver) queue name.
```

Comment:
When I revisit this section, I guess, that we can delete `internal (inside the driver)` from the description, because this is the actual AMQP queue name.

Reply:
This note is needed in order to understand that this queue name is *not public*, available for work from the plugin's API.